### PR TITLE
feat(*): make init/client command work with go module

### LINKED ISSUE #374: `nirvana init` cannot be run outside GOPATH
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

`nirvana init` cannot be run outside GOPATH:
```
$ nirvana init foo
FATAL 1027-18:47:01.921+08 init.go:50 | /tmp/foo is not in GOPATH
```

**What you expected to happen**:

`nirvana init` is able to create a project outside GOPATH.

**How to reproduce it (as minimally and precisely as possible)**:

run `nirvana init` outside GOPATH.

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by iawia002**
**What this PR does / why we need it**:

Continue #371, make `nirvana init/client` work with go module.

```console
$ nirvana init test2
INFO  1027-14:28:38.844+08 init.go:186 | Created project at /Users/bytedance/Downloads/test2

$ cat test2/go.mod
module test2

require (
	github.com/caicloud/nirvana master
)
```

```console
$ nirvana client pkg
INFO  1027-14:33:44.066+08 client.go:83 | Project root directory is /Users/bytedance/Downloads/test2
INFO  1027-14:33:44.069+08 client.go:119 | Generated golang client package test2/client
```

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #374

**Special notes for your reviewer**:

/cc @xpofei @knight42 

**Release note**:

```release-note
Make `nirvana init/client` work with go module.
```

### CODE REVIEW

File: utils/project/path.go [hunk 1]
```diff
@@ -66,11 +66,19 @@ func GoPath(directory string) (goPath string, absPath string, err error) {
 
 // PackageForPath gets package path for a path.
 func PackageForPath(directory string) (string, error) {
-	goPath, absPath, err := GoPath(directory)
+	absPath, err := filepath.Abs(directory)
```
Comment (knight42) [inline utils/project/path.go:69]: Should we update the test `TestPackageForPath`
https://github.com/caicloud/nirvana/blob/f63d17bd5cceeb954650a2a98b38da1ead8b94f8/utils/project/path_test.go#L23
to include the case that `directory` is outside GOPATH?

Comment (iawia002) [reply utils/project/path.go:69]: Updated, PTAL

File: utils/project/path_test.go [hunk 2]
```diff
@@ -21,13 +21,27 @@ import (
...
-	if err != nil {
-		t.Error(err)
+	tests := []struct {
+		name      string
+		directory string
+		wanted    string
+	}{
+		{
+			name:      "in GOPATH test",
+			directory: "./test1",
```
Comment (knight42) [inline utils/project/path_test.go:31]: This case indicates that we run this test inside GOPATH, is it always true?

Comment (iawia002) [reply utils/project/path_test.go:31]: Yeah, something wrong here, if you run this test outside GOPATH, you will get wrong results, I can manually edit the value of "goSrcPaths" to simulate that the directory is in the GOPATH

**Comment by iawia002** [comment]
ping @knight42 any more comments?

**Review by knight42 — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**